diff -r -U 4 PyPubSub-3.3.0/src/pubsub/core/listenerbase.py PyPubSub-3.3.0-patched/src/pubsub/core/listenerbase.py
--- PyPubSub-3.3.0/src/pubsub/core/listenerbase.py	2013-11-17 11:43:52.000000000 +0100
+++ PyPubSub-3.3.0-patched/src/pubsub/core/listenerbase.py	2014-02-11 18:24:58.000000000 +0100
@@ -92,9 +92,12 @@
 
     def __notifyOnDead(self, ref):
         """This gets called when listener weak ref has died. Propagate
         info to Topic)."""
-        notifyDeath = self.__onDead
+        try:
+            notifyDeath = self.__onDead
+        except AttributeError:
+            return
         self._unlinkFromTopic_()
         if notifyDeath is not None:
             notifyDeath(self)
 
