diff -U 4 -r PyPubSub-r343/src/pubsub/core/listenerbase.py PyPubSub-r343-patched/src/pubsub/core/listenerbase.py
--- PyPubSub-r343/src/pubsub/core/listenerbase.py	2013-12-08 13:14:41.000000000 +0100
+++ PyPubSub-r343-patched/src/pubsub/core/listenerbase.py	2013-12-08 13:17:48.000000000 +0100
@@ -92,10 +92,13 @@
 
     def __notifyOnDead(self, ref):
         """This gets called when listener weak ref has died. Propagate
         info to Topic)."""
-        notifyDeath = self.__onDead
         self._unlinkFromTopic_()
+        try:
+            notifyDeath = self.__onDead
+        except AttributeError:
+            pass
         if notifyDeath is not None:
             notifyDeath(self)
 
     def __eq__(self, rhs):
