Index: job.py
===================================================================
--- job.py	(revision 5467)
+++ job.py	(working copy)
@@ -25,7 +25,8 @@
     :param kwargs: dict of keyword arguments to call func with
     :param name: name of the job (optional)
     :param misfire_grace_time: seconds after the designated run time that
-        the job is still allowed to be run
+        the job is still allowed to be run, or 0 if it must always be
+        allowed to run.
     :param coalesce: run once instead of many times if the scheduler determines
         that the job should be run more than once in succession
     :param max_runs: maximum number of times this job is allowed to be
@@ -46,8 +47,8 @@
             raise TypeError('args must be a list-like object')
         if not hasattr(kwargs, '__getitem__'):
             raise TypeError('kwargs must be a dict-like object')
-        if misfire_grace_time <= 0:
-            raise ValueError('misfire_grace_time must be a positive value')
+        if misfire_grace_time < 0:
+            raise ValueError('misfire_grace_time must be a positive value or 0')
         if max_runs is not None and max_runs <= 0:
             raise ValueError('max_runs must be a positive value')
         if max_instances <= 0:
Index: scheduler.py
===================================================================
--- scheduler.py	(revision 5467)
+++ scheduler.py	(working copy)
@@ -282,7 +282,8 @@
         :param name: name of the job
         :param jobstore: stored the job in the named (or given) job store
         :param misfire_grace_time: seconds after the designated run time that
-            the job is still allowed to be run
+            the job is still allowed to be run, or 0 if it must always be
+            allowed to run.
         :type date: :class:`datetime.date`
         :rtype: :class:`~apscheduler.job.Job`
         """
@@ -308,7 +309,8 @@
         :param name: name of the job
         :param jobstore: alias of the job store to add the job to
         :param misfire_grace_time: seconds after the designated run time that
-            the job is still allowed to be run
+            the job is still allowed to be run, or 0 if it must always be
+            allowed to run.
         :rtype: :class:`~apscheduler.job.Job`
         """
         interval = timedelta(weeks=weeks, days=days, hours=hours,
@@ -336,7 +338,8 @@
         :param name: name of the job
         :param jobstore: alias of the job store to add the job to
         :param misfire_grace_time: seconds after the designated run time that
-            the job is still allowed to be run
+            the job is still allowed to be run, or 0 if it must always be
+            allowed to run.
         :return: the scheduled job
         :rtype: :class:`~apscheduler.job.Job`
         """
@@ -451,8 +454,7 @@
             # See if the job missed its run time window, and handle possible
             # misfires accordingly
             difference = datetime.now() - run_time
-            grace_time = timedelta(seconds=job.misfire_grace_time)
-            if difference > grace_time:
+	    if job.misfire_grace_time != 0 and timedelta(seconds=job.misfire_grace_time) < difference:
                 # Notify listeners about a missed run
                 event = JobEvent(EVENT_JOB_MISSED, job, run_time)
                 self._notify_listeners(event)
