Index: __init__.py
===================================================================
--- __init__.py	(.../Release0_72_Branch/taskcoachlib/thirdparty/desktop)	(revision 1979)
+++ __init__.py	(.../Release1_3_Branch/taskcoach/taskcoachlib/thirdparty/desktop)	(revision 3683)
@@ -155,6 +155,9 @@
     elif os.environ.has_key("GNOME_DESKTOP_SESSION_ID") or \
         os.environ.has_key("GNOME_KEYRING_SOCKET"):
         return "GNOME"
+    elif os.environ.has_key('DESKTOP_SESSION') and \
+            os.environ['DESKTOP_SESSION'].lower() == 'lubuntu':
+        return "GNOME"
     elif sys.platform == "darwin":
         return "Mac OS X"
     elif hasattr(os, "startfile"):
